utils/node_local_test.py

```diff
@@ -1457,60 +1457,45 @@ def _helper(self):
 
     return _helper
 
-def needs_dfuse_single(method):
-    """Decorator function for starting dfuse single threaded
-    under posix_tests class"""
-    @functools.wraps(method)
-    def _helper(self):
-        self.dfuse = DFuse(self.server,
-                           self.conf,
-                           caching=True,
-                           pool=self.pool.dfuse_mount_name(),
-                           container=self.container)
-        self.dfuse.start(v_hint=method.__name__, single_threaded=True)
-        try:
-            rc = method(self)
-        finally:
-            if self.dfuse.stop():
-                self.fatal_errors = True
-        return rc
-    return _helper
+class needs_dfuse_with_opt():
+    """Decorator class for starting dfuse under posix_tests class
 
-def needs_dfuse_with_cache(method):
-    """Decorator function for starting dfuse under posix_tests class"""
-    @functools.wraps(method)
-    def _helper(self):
-        self.dfuse = DFuse(self.server,
-                           self.conf,
-                           caching=True,
-                           pool=self.pool.dfuse_mount_name(),
-                           container=self.container)
-        self.dfuse.start(v_hint=method.__name__)
-        try:
-            rc = method(self)
-        finally:
-            if self.dfuse.stop():
-                self.fatal_errors = True
-        return rc
-    return _helper
+    By default runs the method twice, once with caching and once without, however can be
+    configured to behave differently.  Interacts with the run_posix_tests._run_test() method
+    to achieve this.
+    """
 
-def needs_dfuse_no_cache(method):
-    """Decorator function for starting dfuse under posix_tests class"""
-    @functools.wraps(method)
-    def _helper(self):
-        self.dfuse = DFuse(self.server,
-                           self.conf,
-                           caching=False,
-                           pool=self.pool.dfuse_mount_name(),
-                           container=self.container)
-        self.dfuse.start(v_hint=method.__name__)
-        try:
-            rc = method(self)
-        finally:
-            if self.dfuse.stop():
-                self.fatal_errors = True
-        return rc
-    return _helper
+    def __init__(self, caching=None, single_threaded=False):
+        self.caching = caching
+        self.single_threaded = single_threaded
+
+    def __call__(self, method):
+
+        @functools.wraps(method)
+        def _helper(obj):
+
+            caching = self.caching
+            if caching == None:
```

Comment:
(pylint-singleton-comparison) Comparison 'caching == None' should be 'caching is None'